homeassistant/components/switchbee/coordinator.py

```diff
@@ -0,0 +1,84 @@
+"""SwitchBee integration Coordinator."""
+
+from datetime import timedelta
+import logging
+
+from switchbee.api import CentralUnitAPI, SwitchBeeError
+from switchbee.device import DeviceType
+
+from homeassistant.helpers.device_registry import format_mac
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SwitchBeeCoordinator(DataUpdateCoordinator):
```

Comment:
I think this should have the generic type defined:
```suggestion
class SwitchBeeCoordinator(DataUpdateCoordinator[dict[int, SwitchBeeBaseDevice]]):
```